ospd/ospd.py

```diff
@@ -1174,15 +1174,32 @@ def run(self) -> None:
         """ Starts the Daemon, handling commands until interrupted.
         """
 
+        self.scan_collection.data_manager = multiprocessing.Manager()
+
         try:
             while True:
                 time.sleep(SCHEDULER_CHECK_PERIOD)
                 self.scheduler()
                 self.clean_forgotten_scans()
+                self.check_pending_scans()
                 self.wait_for_children()
         except KeyboardInterrupt:
             logger.info("Received Ctrl-C shutting-down ...")
 
+    def check_pending_scans(self):
+        for scan_id in list(self.scan_collection.ids_iterator()):
+            if self.get_scan_status(scan_id) == ScanStatus.PENDING:
+                scan_target = self.scan_collection.scans_table[scan_id].get(
+                    'target'
+                )
```

Comment:
Personally I would write a cleaner interface for the scan collection to hide internal details but that's something for additional improvements
```python
 scan_target = self.scan_collection.get_target_for_scan_id(scan_id)
```

Reply:
I guess there are more places where the data is accessed directly. I will add this to my TODO list.